libp2p/pubsub/gossipsub.py

```diff
@@ -603,6 +586,12 @@ def _get_in_topic_gossipsub_peers_from_minus(
         rpc_msg: bytes = packet.SerializeToString()
 
         # 3) Get the stream to this peer
+        if sender_peer_id not in self.pubsub.peers:
```

Comment:
Feel like this is a "look before you leap" style, the downside is that we have to check every time we access the `self.pubsub.peers`. In a future refactor, raising and catching PeerNotFound exception might look nicer.